src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exception.DukeLoadingException;
+import duke.task.Task;
+
+/**
+ * Manages the operation on the tasks
+ */
+public class TaskList {
+
+    /**
+     * Stores all the task
+     */
+    private ArrayList<Task> tasks;
+    /**
+     * construct a new empty task list
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Loads the list of task from the storage into the task list instance
+     *
+     * @param storedInput - the list of task obtained from storage
+     * @return true if all task loaded successfully else false
+     */
+    public boolean loadTasks(List<String> storedInput) {
+        // parse and store data while looking out for data corruption
+        boolean isCorrupted = false;
+        for (String s : storedInput) {
+            try {
+                Task newTask = Parser.fromStorage(s);
+                this.tasks.add(newTask);
+            } catch (DukeLoadingException | DateTimeParseException e) {
+                isCorrupted = true;
+            }
+        }
+        return isCorrupted;
+    }
+
+    /**
+     * returns the current task list length
```

Comment:
Might be nicer if the first letter of the first word of the JavaDoc comment is capitalised. (e.g. "Returns...")